src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/AspNetCoreHostingDiagnosticListener.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using Microsoft.ApplicationInsights.AspNetCore.Extensions;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DiagnosticAdapter;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+namespace Microsoft.ApplicationInsights.AspNetCore.DiagnosticListeners
+{
+    internal class AspNetCoreHostingDiagnosticListener: IApplicationInsightDiagnosticListener
+    {
+        private readonly TelemetryClient _client;
```

Comment:
nitpick: Sorry to be pedantic, I did ask in the meeting a couple weeks ago that the new code have doc comments and be StyleCop (defaults) compliant.  I would really appreciate that, thank-you. :)